src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy

```diff
@@ -17,15 +17,32 @@ import org.sonatype.nexus.ci.iq.PolicyEvaluationHealthAction
 import org.sonatype.nexus.ci.iq.PolicyEvaluationProjectAction
 
 import jenkins.model.Jenkins
+import groovy.json.JsonBuilder
 import lib.JenkinsTagLib
 import lib.LayoutTagLib
 
+// number of last recent builds
+final MAX_BUILDS_TO_GRAPH = 5;
+
 def t = namespace(JenkinsTagLib.class)
 def l = namespace(LayoutTagLib.class)
 
 def projectAction = (PolicyEvaluationProjectAction) it
 def actions = projectAction.getJob().lastCompletedBuild.getActions(PolicyEvaluationHealthAction.class)
 
+def policyEvaluations = projectAction.getJob().getBuilds().stream()
+    .filter{!it.isBuilding()}
+    .map{
+      def healthActions = it.getActions(PolicyEvaluationHealthAction.class)
+      // getActions returns only 1 or empty collection, see PolicyEvaluationHealthAction#getProjectActions
+      return healthActions ? healthActions.get(0) : null
```

Comment:
0 : won't returning the null here cause the subsequent filter or map to throw an NPE?

Reply:
why? on the next step I filter all null values to prevent NPE:
`.filter{Objects.nonNull(it)}`